File: server/internal/database/types/avatar.go
package types

import "scrumlr.io/server/internal/database/types/avatar"

type Avatar struct {
	AccentColorClass string                 `json:"accentColorClass,omitempty"`
	AccessoriesType  avatar.AccessoriesType `json:"accessoriesType,omitempty"`
	ClotheColor      avatar.ClotheColor     `json:"clotheColor,omitempty"`
	ClotheType       avatar.ClotheType      `json:"clotheType,omitempty"`
	EyeType          avatar.EyeType         `json:"eyeType,omitempty"`
	EyebrowType      avatar.EyebrowType     `json:"eyebrowType,omitempty"`
	FacialHairColor  avatar.FacialHairColor `json:"facialHairColor,omitempty"`
	FacialHairType   avatar.FacialHairType  `json:"facialHairType,omitempty"`
	GraphicType      avatar.GraphicType     `json:"graphicType,omitempty"`
	HairColor        avatar.HairColor       `json:"hairColor,omitempty"`
	MouthType        avatar.MouthType       `json:"mouthType,omitempty"`
	SkinColor        avatar.SkinColor       `json:"skinColor,omitempty"`
	TopType          avatar.TopType         `json:"topType,omitempty"`
}

File: src/types/avatar.ts
import {AccessoriesType, ClotheType, EyebrowType, EyeType, FacialHairType, GraphicType, HairColorName, MouthType, SkinColor, TopType} from "@gamepark/avataaars";
import FacialHairColorName from "@gamepark/avataaars/src/avatar/top/facialHair/FacialHairColorName";
import ClotheColorName from "@gamepark/avataaars/src/avatar/clothes/ClotheColorName";
import {Color} from "constants/colors";

export interface AvataaarProps {
  // possibly look into separating accentColor from the other properties at some point
  // otherwise don't forget to omit when needed
  backgroundColor: Color;
  skinColor: SkinColor;
  topType: TopType;
  clotheColor: ClotheColorName;
  graphicType: GraphicType;
  clotheType: ClotheType;
  hairColor: HairColorName;
  facialHairColor: FacialHairColorName;
  facialHairType: FacialHairType;
  accessoriesType: AccessoriesType;
  eyeType: EyeType;
  eyebrowType: EyebrowType;
  mouthType: MouthType;
}

export type DisabledOnType = Partial<{[K in keyof AvataaarProps]: AvataaarProps[K][]}>;

export type AvatarGroup = {
  values: ReadonlyArray<string>;
  key: keyof AvataaarProps;
  disabledOn?: DisabledOnType;
};

export type AvatarConfig = Record<string, ReadonlyArray<AvatarGroup>>;
